src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -18,14 +20,30 @@ using namespace mapbox::geojsonvt;
 namespace mbgl {
 namespace style {
 namespace conversion {
+
+struct ToFeatureCollection {
+    mapbox::geojson::feature_collection operator()(const mapbox::geojson::feature_collection value) const {
+        return value;
+    }
+    mapbox::geojson::feature_collection operator()(const mapbox::geojson::feature value) const {
+        return { value };
+    }
+    mapbox::geojson::feature_collection operator()(const mapbox::geojson::geometry value) const {
+        return { { value } };
+    }
+};
+
 template <>
 Result<GeoJSON> convertGeoJSON(const JSValue& value) {
     Options options;
     options.buffer = util::EXTENT / util::tileSize * 128;
     options.extent = util::EXTENT;
 
     try {
-        return GeoJSON { std::make_unique<GeoJSONVT>(Convert::convert(value, 0), options) };
+        ToFeatureCollection toFeatureCollection;
+        const auto geojson = mapbox::geojson::convert(value);
+        const auto features = apply_visitor(toFeatureCollection, geojson);
+        return GeoJSON { std::make_unique<GeoJSONVT>(features, options) };
```

Comment:
It's awkward to be unable to directly compose `mapbox::geojson::convert` and `mapbox::GeoJSONVT::GeoJSONVT `. We should add a `GeoJSONVT` constructor that accepts `mapbox::util::variant<geometry, feature, feature_collection>`. cc @mourner